pkg/download/download.go

```diff
@@ -130,6 +142,24 @@ func Download(ctx context.Context, endpoint string, opts ...Option) (b []byte, e
 func download(req *http.Request, dlOpts *downloadOptions) (data []byte, err error) {
 	client := &http.Client{}
 
+	if dlOpts.LowSrcPort {
+		port := 100 + rand.Intn(512)
+
+		localTCPAddr, tcperr := net.ResolveTCPAddr("tcp", ":"+strconv.Itoa(port))
+		if tcperr != nil {
+			return nil, retry.ExpectedError(fmt.Errorf("resolving source tcp address: %s", tcperr.Error()))
+		}
+
+		client = &http.Client{
+			Transport: &http.Transport{
```

Comment:
the worst thing which I hate a lot in Go, is that actually `nil` `Transport` is not an empty transport - there's a lot of good defaults set explicitly. we need to preserve them somehow

Reply:
we should actually be using https://github.com/hashicorp/go-cleanhttp here to build "clean" client and tweak things after that

we could handle port reuse in the dialer